src/sdlwindow.cpp

```diff
@@ -239,6 +244,23 @@ void sdlwindow_update( void )
 			SDL_HideWindow( g_SDLWindow );
 		}
 	}
+
+	if ( g_bUpdateSDLWindowTitle == true )
+	{
+		g_bUpdateSDLWindowTitle = false;
+		SDL_SetWindowTitle( g_SDLWindow, g_SDLWindowTitle.c_str() );
+	}
+}
+
+void sdlwindow_title( const char* title )
+{
+	title = title ? title : DEFAULT_TITLE;
+	if ( g_SDLWindowTitle != title )
+	{
+		g_SDLWindowTitle = title ? title : DEFAULT_TITLE;
+		g_bUpdateSDLWindowTitle = true;
```

Comment:
This is racy, e.g. if the window title is updated fast enough. Need a lock to properly synchronize accesses to globals.

We're not good at dealing with races in general, I think the existing code just hopes for the best.

Reply:
I don't have a 100% view of what global is accessed from where, so i assumed sdlwindow_update would never be called while sdlwindow_title is running.

I pushed an update where i use an atomic bool for the update check and a mutex for the title string, please take a look as i'm not very familiar with C++'s parallelism APIs.

Reply:
I _think_ this is working correctly with the default sequentially-consistent ordering (ref https://en.cppreference.com/w/cpp/atomic/memory_order#Sequentially-consistent_ordering). I'd also be fine with just using a mutex without an atomic.

Reply:
Truth be told, i added the mutex after i realized the atomic bool isn't enough. I pushed another update that removes the atomic bool and puts everything between the mutex lock/unlock as the atomic bool isn't really necessary.